### PR TITLE
History lens part 1

### PR DESCRIPTION
**Pull request by Yanpas**
async constructors
log command
process spawn small fix

### CODE REVIEW

File: src/model.ts [hunk 1]
```diff
@@ -317,14 +326,14 @@ export class Model implements IDisposable {
     const mm = new Minimatch("*");
     const newLevel = level + 1;
     if (newLevel <= this.maxDepth) {
-      fs.readdirSync(path).forEach(file => {
+      await fs.readdirSync(path).forEach(async file => {
```
Comment (JohnstonCode) [inline src/model.ts]: await a synchronous call?

File: src/svnRepository.ts [hunk 2]
```diff
@@ -3,32 +3,50 @@ import * as path from "path";
...
+import { parseSvnLog } from "./logParser";
 import { parseStatusXml } from "./statusParser";
 import { Svn } from "./svn";
-import { fixPathSeparator } from "./util";
+import { fixPathSeparator, unwrap } from "./util";
 
 export class Repository {
-  private _info: { [index: string]: ISvnInfo } = {};
+  private _infos: { [index: string]: ISvnInfo } = {};
+  private _info?: ISvnInfo;
```
Comment (JohnstonCode) [inline src/svnRepository.ts]: Better naming maybe? what's the difference between info infos?

**Comment by JohnstonCode** [comment]
Also need to fix `src/common/types.ts:97:3 - error TS2300: Duplicate identifier 'Info'.`

**Comment by JohnstonCode** [comment]
Do like the async constructors btw

**Comment by Yanpas** [comment]
Fixed

File: src/repository.ts [hunk 3]
```diff
@@ -186,9 +193,13 @@ export class Repository {
...
 
+    // TODO on svn switch event fired two times since two files were changed
     onRelevantSvnChange(
-      this._onDidChangeRepository.fire,
-      this._onDidChangeRepository,
+      async (e: Uri) => {
+        await this.repository.updateInfo();
+        this._onDidChangeRepository.fire(e);
+      },
+      this,
```
Comment (Yanpas) [inline src/repository.ts]:
```suggestion
      undefined,
```
this is not mandatory since lambda doesn't have it's own this

**Pull request merged**